scripts/Muon/GUI/Common/phase_table_widget/phase_table_presenter.py

```diff
@@ -70,49 +73,75 @@ def create_calculation_thread(self):
-                                                             self.handle_calculation_success,
-                                                             self.handle_calculation_error)
+            self.phasequad_calculation_thread = self.create_phase_quad_calculation_thread()
 
-        self.phasequad_calculation_thread.start()
+            self.phasequad_calculation_thread.threadWrapperSetUp(self.handle_calculation_started,
+                                                                 self.handle_phasequad_calculation_success,
+                                                                 self.handle_calculation_error)
+
+            self.phasequad_calculation_thread.start()
 
     def create_phase_quad_calculation_thread(self):
         self._phasequad_calculation_model = ThreadModelWrapper(self.calculate_phase_quad)
         return thread_model.ThreadModel(self._phasequad_calculation_model)
 
     def calculate_phase_quad(self):
-        parameters = self.get_parameters_for_phase_quad()
-        phasequad_workspace_name = get_phase_quad_workspace_name(parameters['InputWorkspace'], parameters['PhaseTable'])
-
-        self.current_alg = mantid.AlgorithmManager.create("PhaseQuad")
-        phase_quad = run_PhaseQuad(parameters, self.current_alg, phasequad_workspace_name)
-        self.current_alg = None
-
-        self.add_phase_quad_to_ADS(parameters['InputWorkspace'], phase_quad)
-
-    def get_parameters_for_phase_quad(self):
-        parameters = {}
-        if self.context.phase_context.options_dict['phase_table_for_phase_quad'] == 'Construct':
-            parameters['PhaseTable'] = self.calculate_phase_table()
-        else:
-            parameters['PhaseTable'] = self.context.phase_context.options_dict['phase_table_for_phase_quad']
-
-        parameters['InputWorkspace'] = self.context.phase_context.options_dict['phase_quad_input_workspace']
+        self.context.group_pair_context.add_phasequad(self._phasequad_obj)
 
-        return parameters
+        self.context.calculate_phasequads(
+            self._phasequad_obj.name, self._phasequad_obj)
+        self.phase_quad_calculation_complete_notifier.notify_subscribers(
+            self._phasequad_obj.Re.name)
+        self.phase_quad_calculation_complete_notifier.notify_subscribers(
+            self._phasequad_obj.Im.name)
 
-    def add_phase_quad_to_ADS(self, input_workspace, phasequad_workspace_name):
-        run = re.search('^{}([0-9, -]+)[;,_]?'.format(self.context.data_context.instrument), input_workspace).group(1)
+    def handle_phasequad_calculation_success(self):
+        self.enable_editing_notifier.notify_subscribers()
+        self.view.enable_widget()
+        self.view.disable_cancel()
 
-        directory = get_base_data_directory(self.context, run)
+        names = [self._phasequad_obj.Re.name, self._phasequad_obj.Im.name]
+        self.current_alg = None
+        pair_added = True  # if state == 2 else False
```

Comment:
I can't see how pair_added can be false here?

Reply:
at present it cannot. I left this in for when we have a table, with the analyse tick box.